src/linalg.jl

```diff
@@ -1435,13 +1435,19 @@ kron(A::Union{SparseVector,AbstractSparseMatrixCSC,AdjOrTrans{<:Any,<:AbstractSp
 kron(A::VecOrMat, B::Union{SparseVector,AbstractSparseMatrixCSC,AdjOrTrans{<:Any,<:AbstractSparseMatrixCSC}}) =
     kron(sparse(A), B)
 
-# sparse vec/mat ⊗ Diagonal and vice versa
-Base.@propagate_inbounds kron!(C::SparseMatrixCSC, A::Diagonal{T}, B::Union{SparseVector{S}, AbstractSparseMatrixCSC{S}}) where {T<:Number, S<:Number} = kron!(C, sparse(A), B)
+# sparse vec/mat ⊗ Diagonal etc. and vice versa
+const MorallySparse1{T} = Union{Bidiagonal{T}, Diagonal{T}, SymTridiagonal{T}, Tridiagonal{T}}
+const MorallySparse{T} = Union{MorallySparse1{T}, Adjoint{T, <:MorallySparse1}, Transpose{T, <:MorallySparse1}}
```

Comment:
For pretty much every reasonable number type (in symmetric cases only for `Real`), `adjoint` and `transpose` yield structured matrix types instead of `Adjoint` or `Transpose` wrappers. In other cases, like `adjoint(Symtridiagonal{Complex})` we may not have fast `sparse` constructors. So, I believe we can safely omit the `Adjoint/Transpose` stuff here, and maybe call it `const StructuredMatrices` or `const Diagonals`? Though the latter may be challenging to distinguish from `Diagonal`. We have `LinearAlgebra.BiTriSym` already BTW.

Reply:
Done.